lib/evaluators/AsyncEvaluator.ts

```diff
@@ -1,23 +1,30 @@
 import * as RDF from 'rdf-js';
-import { Algebra as Alg } from 'sparqlalgebrajs';
+import {NamedNode} from 'rdf-js';
+import {Algebra as Alg} from 'sparqlalgebrajs';
 
 import * as E from '../expressions/Expressions';
 
-import { transformAlgebra } from '../Transformation';
-import { Bindings, ExpressionEvaluator } from '../Types';
+import {transformAlgebra} from '../Transformation';
+import {Bindings, ExpressionEvaluator} from '../Types';
 
-import { AsyncRecursiveEvaluator } from './RecursiveExpressionEvaluator';
+import {AsyncRecursiveEvaluator} from './RecursiveExpressionEvaluator';
+import {SyncExtensionFunctionCB} from './SyncEvaluator';
 
 type Expression = E.Expression;
 type Term = E.TermExpression;
 
+export type AsyncExtensionFunction = (args: RDF.Term[]) => Promise<RDF.Term>;
+export type AsyncExtensionFunctionCB = (functionNamedNode: NamedNode) => AsyncExtensionFunction;
+
 export interface AsyncEvaluatorConfig {
   now?: Date;
   baseIRI?: string;
 
   exists?: (expression: Alg.ExistenceExpression, mapping: Bindings) => Promise<boolean>;
   aggregate?: (expression: Alg.AggregateExpression) => Promise<RDF.Term>;
   bnode?: (input?: string) => Promise<RDF.BlankNode>;
+  asyncExtensionFunctionCB?: AsyncExtensionFunctionCB;
```

Comment:
```suggestion
  extensionFunctionCallback?: AsyncExtensionFunctionCB;
```
(same name in `SyncEvaluatorConfig`)